### ISSUE TITLE
Implement synchronization with Acoustic

### ISSUE THREAD

**Issue by jwhitlock** [mozilla-it/ctms-api#53]
The Postgres Schema doc has the ``acoustic_sync`` table, used to track syncing with Acoustic:

```sql

CREATE TABLE acoustic_sync (
	id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
	table_name VARCHAR(50) NOT NULL,
	row_id INT NOT NULL,
	change_timestamp TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT NOW(),
	updated_in_acoustic BOOLEAN DEFAULT false,
	UNIQUE (table_name, row_id, updated_in_acoustic)
);
```

This may be useful for the direct-to-DB import, or could be done as part of adding an Acoustic interface.

Tracked in Jira:OSS-282 (Q1 Sprint 5) and Jira:OSS-303 (~Q1 Sprint 6~ Q2 Sprint 2) and Jira:OSS-398 (Q2 Sprint 3)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/CTMS-54)

**Comment by bsieber-mozilla** [comment]
Hmm, I'm a bit confused with this table; will this table ever be cleaned or will it be expanding ad infinitum? If there's no cleaning then I'd be concerned with the long-term performance of this (unless indexes can help us tremendously here, but I'm not too well versed with postgres and don't know how much that could help us).

**Comment by bsieber-mozilla** [comment]
(Found this referenced in PT1: https://jira.mozilla.com/browse/OSS-282)

**Comment by jwhitlock** [comment]
Implemented in:

* PR #181 - READY: Acoustic pending updates - Basic Operations + Scheduling of PendingRecords 
* PR #185 - Add basics of acoustic sync background job 
* PR #193 - Acoustic Sync - From Ctms/DB to Acoustic 
* PR #202 - Adding ability to use env var for retry limit
* PR #207 - Adding FeatureFlags to the Sync Job as Background EnvVars

The design discussions took place outside of this issue, and evolved differently than the designs @cvalaas and I hashed out above. See the PRs for the actual implementation.

@bsieber-mozilla I think we can close this issue and switch to bug fix mode. Are there more features you want to implement before closing?
